services/weather.js

```diff
@@ -1,27 +1,101 @@
 const fetch = require('node-fetch');
-const { urlBuilder, formatWeatherData } = require('../helpers/weather');
+const Weather = require('../models/Weather');
+const { urlBuilder, formatWeatherData, getToday } = require('../helpers/weather');
 
 /**
- * Fetching data from open waether API
+ * Fetching data from open weather API
  * @route /weather/:city
  * @param string city
  */
 module.exports.fetchWeatherDataByCity = async (city) =>
 {
     try
     {
-        const weatherJsonData = await fetchWeatherData(city);
+        const today = getToday();
+
+        // check from db first
+        let weatherJsonData = await Weather.findOne({ city, date: today });
+
+        // if there is not data for that then loading from third party service
+        // then insert in to db
+        if (!weatherJsonData)
+        {
+            console.log("load from api")
+            weatherJsonData = await fetchWeatherDataFromAPI(city);
+
+            this.insertWeatherDataByCity(weatherJsonData);
+        } else
+        {
+            console.log(`load weather data from db`);
+        }
         return weatherJsonData;
     } catch (error)
     {
         console.log(error)
+        return error;
     }
 };
 
-const fetchWeatherData = async (city) =>
+/**
+ * Fetch data from API and then format the data
+ * @param {string} city 
+ * @returns array
+ */
+const fetchWeatherDataFromAPI = async (city) =>
 {
     const endpoint = urlBuilder(city);
     const apiResponse = await fetch(endpoint);
     const weatherJsonData = await apiResponse.json();
+
     return formatWeatherData(weatherJsonData);
 };
+
+
+/**
+ * Insert data from open waether API
+ * @route /weather/:city
+ * @param string city
+ */
+module.exports.insertWeatherDataByCity = async (weatherData) =>
+{
+    try
+    {
+        console.log(`service insert weather data`);
+        return Weather.create(weatherData);
+    } catch (error)
+    {
+        return error
+    }
+};
+
+/**
+ * Fetching data of all city by date
+ * @route /weather/:city
+ * @param string city
+ */
+module.exports.fetchWeatherDataByDate = async (date = null) =>
+{
+    try
+    {
+        const searchDay = (!date) ? getToday() : date;
+
+        // check from db
+        let weatherJsonData = await Weather.find({ date: searchDay });
+
+        if (!weatherJsonData)
+        {
+            console.log(`There is not any records matched with day`)
+            // TODO: NOTFOUND exceptions?
+            return false;
```

Comment:
create custom error `NotFoundException` for not-found cases and use it everywhere: https://javascript.info/custom-errors